llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -131,6 +131,12 @@ bool AMDGPUInstructionSelector::selectCOPY(MachineInstr &I) const {
   Register SrcReg = Src.getReg();
 
   if (isVCC(DstReg, *MRI)) {
+    if (SrcReg.isPhysical() && SrcReg != AMDGPU::SCC) {
+      const TargetRegisterClass *DstRC = MRI->getRegClassOrNull(DstReg);
+      if (DstRC)
+        return DstRC->contains(SrcReg);
+    }
```

Comment:
Can you split this into a separate change, with MIR tests? I suspect you can just directly return true.

Reply:
This should be split to a separate change